pkg/oc/cli/cmd/push/binary.go

```diff
@@ -0,0 +1,368 @@
+package push
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapiv1 "k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	imageapi "github.com/openshift/origin/pkg/image/apis/image"
+	"github.com/openshift/origin/pkg/image/apis/image/docker10"
+	"github.com/openshift/origin/pkg/image/apis/image/v1"
+	clientset "github.com/openshift/origin/pkg/image/generated/clientset/typed/image/v1"
+)
+
+var (
+	pushBinaryLong = templates.LongDesc(`
+		Create a new image by pushing an archive
+
+		This command assists users in creating new images that can be used as part of builds
+		or as deployed applications. It accepts a zip or tar.gz file that will become a new 
+		image in an image stream. You may also specify image metadata like the entrypoint, 
+		environment variables, or labels to create a runnable image.
+
+		Experimental: This command is under active development and may change without notice.`)
+
+	pushBinaryExample = templates.Examples(`
+		# Upload myblob.zip as a new image in the app:binary image stream tag
+	  $ %[1]s binary --to=app:binary myblob.zip
```

Comment:
Hrm - I'd say in openshift the verb "push" belongs most strongly to images.  So in this case push implies image, and binary is the chunk of thing you're pushing.  So like binary build today, we'd have the equivalents to `--from-repo`, `--from-file`, and `--from-dir`.  This is more like `--from-file`.